Trellix Endpoint Security HX Connector

SharmilaMS-Hcl opened the pull request:
Added connector files for Trellix Endpoint Security HX data source

SharmilaMS-Hcl committed:
Trellix Endpoint Security HX Connector

Added connector files for Trellix Endpoint Security HX data source

SharmilaMS-Hcl committed:
Merge branch 'opencybersecurityalliance:develop' into trellix_hx_v1

SharmilaMS-Hcl committed:
updated readme

SharmilaMS-Hcl committed:
Merge branch 'trellix_hx_v1' of https://github.com/SharmilaMS-Hcl/stix-shifter into trellix_hx_v1

SharmilaMS-Hcl committed:
Merge branch 'develop' into trellix_hx_v1

SharmilaMS-Hcl committed:
Merge branch 'develop' into trellix_hx_v1

LarsenEric commented on stix_shifter_modules/trellix_endpoint_security_hx/README.md:
```diff
@@ -0,0 +1,697 @@
+  are specified
+- Based on the concurrent search limit of the data source, the concurrent search limit of the connector should be 
+  less than or equal to the data source. 
+- A maximum of 15 searches only can be created in data source. In order to create more searches, the existing 
+  search ids created needs to be deleted.
```
Is there an error message the user will receive for this scenario?

SharmilaMS-Hcl replied:
User will receive the below error for this scenario
{
    "success": false,
    "connector": "trellix_endpoint_security_hx",
    "error": "trellix_endpoint_security_hx connector error => Exceeded limit on existing searches",
    "code": "service_unavailable"
}

LarsenEric replied:
This should be more specific about what the max is if the user hits it. They should know exactly why it is failing without needing to dig for an answer.

SharmilaMS-Hcl replied:
Detailed message related to number of max limit hits has been added to the message.   Error Message : The total search limit - {limit number}  is reached. Delete an existing search to create a new one

LarsenEric commented on stix_shifter_modules/trellix_endpoint_security_hx/README.md:
```diff
@@ -0,0 +1,697 @@
+
+- A maximum of 25 conditions using AND operator can be specified. Exception will be raised if more than 25 conditions 
+  are specified
+- Based on the concurrent search limit of the data source, the concurrent search limit of the connector should be 
+  less than or equal to the data source. 
```
Is there an error message the user will receive for this scenario?

SharmilaMS-Hcl replied:
User will receive the below error for this scenario, 
{
"success": false,
"connector": "trellix_endpoint_security_hx",
"error": "trellix_endpoint_security_hx connector error => Exceeded limit on existing searches",
"code": "service_unavailable"
}

LarsenEric replied:
This should be more specific about what limits are causing it to fail if the user hits it. They should know exactly why it is failing without needing to dig for an answer.

SharmilaMS-Hcl replied:
The error message is updated as follows:  The total search limit - {limit number} is reached. Delete an existing search to create a new one.

LarsenEric commented on stix_shifter_modules/trellix_endpoint_security_hx/README.md:
```diff
@@ -0,0 +1,697 @@
+
+### Limitations
+
+- A maximum of 25 conditions using AND operator can be specified. Exception will be raised if more than 25 conditions 
+  are specified
```
Is there an error message the user will receive for this scenario?

SharmilaMS-Hcl replied:
the user will receive below message for this scenario
{
    "success": false,
    "connector": "trellix_endpoint_security_hx",
    "error": "trellix_endpoint_security_hx connector error => The maximum number of search terms is 25",
    "code": "invalid_query"
}

LarsenEric commented on stix_shifter_modules/trellix_endpoint_security_hx/README.md:
```diff
@@ -0,0 +1,697 @@
+  less than or equal to the data source. 
+- A maximum of 15 searches only can be created in data source. In order to create more searches, the existing 
+  search ids created needs to be deleted.
+- As per datasource limitation, If a host set containing more than 1000 hosts, the 1000 hosts that responds first
+  will be returned in response. Configure multiple host sets in the datasource to fetch records from more than 1000 hosts.
```
```suggestion
- As per the FireEye data source limitation, if a host set contains more than 1000 hosts, the 1000 hosts that respond first will be returned in the response. To avoid this scenario, configure multiple host sets in the data source to fetch the records from more than 1000 hosts.
```

SharmilaMS-Hcl replied:
updated as per suggestion

LarsenEric commented on stix_shifter_modules/trellix_endpoint_security_hx/stix_translation/json/operators.json:
```diff
@@ -0,0 +1,13 @@
+    "ComparisonComparators.In": "equals",
+    "ComparisonComparators.Matches": "contains",
+    "ObservationOperators.Or": "or",
+    "ObservationOperators.And": "or"
```
This seems incorrect to me. 

> "ObservationOperators.And": "or"

This seems to be a contradiction, setting an AND operator as an OR operator

SharmilaMS-Hcl replied:
It is as designed. AND operator between two observation statements is converted to OR. All the connectors are following the same.

LarsenEric commented:
I notice that everything references Trellix but in the ReadMe references FireEye. Can you please be consistent with the naming and ensure it is properly conveyed.

LarsenEric commented on stix_shifter_modules/trellix_endpoint_security_hx/stix_translation/json/from_stix_map.json:
```diff
@@ -0,0 +1,77 @@
+      "name": ["File Name"],
+      "size": ["Size in bytes"],
+      "hashes.MD5": ["File MD5 Hash"],
+      "parent_directory_ref.path": ["File Full Path"],
+      "x_path": ["File Full Path"]
```
is there a need for both of these if they are both in fact the "File Full Path"?

SharmilaMS-Hcl replied:
parent_directory_ref.path actually searches for directory path of the file, and x_path searches for full path of the file including the file name.  The custom field x_path field is added based on the [stix extensions](https://github.com/opencybersecurityalliance/stix-extensions/blob/main/2.x/SCO/custom-file.md)

LarsenEric commented on stix_shifter_modules/trellix_endpoint_security_hx/stix_translation/json/from_stix_map.json:
```diff
@@ -0,0 +1,77 @@
+  },
+  "directory": {
+    "fields": {
+      "path": ["File Full Path"]
```
if this is a directory not a file that should be updated to reflect that.

SharmilaMS-Hcl replied:
Trellix doesnt have separate field for directory path of the file. The file full path which is mapped holds the value of  directory  of the file and file name

LarsenEric commented on stix_shifter_modules/trellix_endpoint_security_hx/stix_translation/json/stix_2_1/from_stix_map.json:
```diff
@@ -0,0 +1,73 @@
+  },
+  "directory": {
+    "fields": {
+      "path": ["File Full Path"]
```
Should this be file or directory?

SharmilaMS-Hcl replied:
same as above.  this is the directory of the file. Trellix doesnt have separate field for directory path of the file. The file full path which is mapped holds the value of directory of the file and file name

SharmilaMS-Hcl committed:
Merge branch 'develop' into trellix_hx_v1

SharmilaMS-Hcl committed:
Merge branch 'develop' into trellix_hx_v1

SharmilaMS-Hcl committed:
updated review comments

updated suggestions in read me file

SharmilaMS-Hcl commented:
> I notice that everything references Trellix but in the ReadMe references FireEye. Can you please be consistent with the naming and ensure it is properly conveyed.

Updated as per suggestions and maintained the name trellix across the read me file.

SharmilaMS-Hcl committed:
Updated the code related to Exception

Updated the code related to Exceeded limit exception.
added the mapping of parent file name in to-stix files.

DerekRushton committed:
Merge branch 'develop' into trellix_hx_v1
